great_expectations/data_context/util.py

```diff
@@ -205,7 +248,208 @@ def substitute_config_variable(
+        raise ValueError(
+            f"Could not match the value with regex {AWS_SECRET_MANAGER_SLOW_REGEX}"
+        )
+
+    region_name = matches.group(1)
+    secret_name = matches.group(3)
+    secret_version = matches.group(4)
+    secret_key = matches.group(5)
+
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager", region_name=region_name)
+
+    secret_response = client.get_secret_value(
+        SecretId=secret_name, VersionId=secret_version
+    )
+    # Decrypts secret using the associated KMS CMK.
+    # Depending on whether the secret is a string or binary, one of these fields will be populated.
+    if "SecretString" in secret_response:
+        secret = secret_response["SecretString"]
+    else:
+        secret = base64.b64decode(secret_response["SecretBinary"]).decode("utf-8")
+    if secret_key:
+        secret = json.loads(secret)[secret_key]
+    return secret
+
+
+def substitute_value_from_gcp_secret_manager(value):
+    """
+    This methods uses a google.cloud.secretmanager.SecretManagerServiceClient to try to retrieve the secret value
+    from the elements it is able to parse from the input value.
+
+    value: string with pattern ``secret|projects/${project_id}/secrets/${secret_name}``
+
+        optional : after the value above, a secret version can be added ``/version/${secret_version}``
+
+        optional : after the value above, a secret key can be added ``|${secret_key}``
+
+    - project_id: `Project ID of the GCP project on which the secret manager is implemented <https://cloud.google.com/resource-manager/docs/creating-managing-projects#before_you_begin>`_
+    - secret_name: Name of the secret
+    - secret_version: ID of the version of the secret
+    - secret_key: Only if the secret's data is a JSON string, which key of the dict should be retrieve
+
+    :param value: a string that matches the following regex ``^secret|projects/[a-z0-9_-]{6,30}/secrets``
+
+    :return: a string with the value substituted by the secret from the GCP Secret Manager store
+    :raises: ImportError, ValueError
+    """
+    if not secretmanager:
+        logger.error(
+            "secretmanager is not installed, please install great_expectations with gcp extra > "
+            "pip install great_expectations[gcp]"
+        )
+        raise ImportError("Could not import secretmanager from google.cloud")
+
+    client = secretmanager.SecretManagerServiceClient()
+    matches = GCP_SECRET_MANAGER_SLOW_REGEX.match(value)
+
+    if not matches:
+        raise ValueError(
+            f"Could not match the value with regex {GCP_SECRET_MANAGER_SLOW_REGEX}"
+        )
+
+    project_id = matches.group(1)
+    secret_id = matches.group(2)
+    secret_version = matches.group(3)
+    secret_key = matches.group(4)
+    if not secret_version:
+        secret_version = "latest"
+    name = f"projects/{project_id}/secrets/{secret_id}/version/{secret_version}"
```

Comment:
```suggestion
    name = f"projects/{project_id}/secrets/{secret_id}/versions/{secret_version}"
```

- Should this be `versions`?

Reply:
Indeed, it should be versions, well spotted !